### PR TITLE
use native window opener functionality

### PR DESCRIPTION
**Pull request by bridiver**
There are still some electron tests that are failing, but they don't affect us so I will come back to them later. This requires a few small changes in browser-laptop before it should be merged

### CODE REVIEW

**Comment by bridiver** [comment]
this fixes browser-laptop issues #98, #97, #57 and #49

File: atom/browser/api/atom_api_app.cc [hunk 1]
```diff
@@ -225,6 +226,42 @@ void App::OnLogin(LoginHandler* login_handler) {
...
+
+  if (container_type == WINDOW_CONTAINER_TYPE_BACKGROUND) {
+    return true;
+  }
+
+  // this will override allowpopups
+  // we need some way to integerate
+  // it so we can turn popup blocking
+  // off if desired
+  if (!user_gesture) {
```
Comment (bbondy) [inline atom/browser/api/atom_api_app.cc]: Super minor nit: capital on first character of a comment and you can probably make this comment only span 2 lines and still stay under the char count.

File: atom/browser/lib/guest-view-manager.coffee [hunk 2]
```diff
@@ -150,11 +155,28 @@ destroyGuest = (embedder, id) ->
...
+  event.sender.emit 'new-window', event, url, frameName, disposition, options
+  if (event.sender.isGuest() and not event.sender.allowPopups) or event.defaultPrevented
+    event.returnValue = null
+  else
+    event.returnValue = addGuest(event.sender, createWebContents(event.sender, options))
+
+process.on 'ATOM_SHELL_GUEST_VIEW_MANAGER_NEXT_INSTANCE_ID', (event) ->
+  event.returnValue = getNextInstanceId event.sender
+
+process.on 'ATOM_SHELL_GUEST_VIEW_MANAGER_REGISTER_GUEST', (event, webContents, id) ->
```
Comment (bridiver) [inline atom/browser/lib/guest-view-manager.coffee]: I should probably add some comments for these (register, add, create, attach)

File: atom/browser/api/atom_api_web_contents.cc [hunk 3]
```diff
@@ -323,16 +319,121 @@ bool WebContents::ShouldCreateWebContents(
...
+void WebContents::ResumeLoadingCreatedWebContents() {
+  if (delayed_load_url_params_.get()) {
+    GetWebContents()->GetController().LoadURLWithParams(
+                                              *delayed_load_url_params_.get());
+    delayed_load_url_params_.reset(nullptr);
+  }
+  // do we need this?
+  // GetRenderViewHost()->GetProcess()->ResumeRequestsForView(route_id);
+  // GetRenderViewHost()->GetProcess()->ResumeRequestsForView(
+  //                                                       main_frame_route_id);
```
Comment (bbondy) [inline atom/browser/api/atom_api_web_contents.cc]: nit: Remove or find out if we actually need it.

File: atom/browser/atom_browser_client.cc [hunk 4]
```diff
@@ -101,6 +101,44 @@ AtomBrowserClient::AtomBrowserClient() : delegate_(nullptr) {
...
+                                      WindowOpenDisposition disposition,
+                                      const blink::WebWindowFeatures& features,
+                                      bool user_gesture,
+                                      bool opener_suppressed,
+                                      content::ResourceContext* context,
+                                      int render_process_id,
+                                      int opener_render_view_id,
+                                      int opener_render_frame_id,
+                                      bool* no_javascript_access) {
+  // do we need the delegate here?
```
Comment (bbondy) [inline atom/browser/atom_browser_client.cc]: nit: remove or find out if we actually need it here.

File: atom/renderer/lib/inspector.coffee [hunk 5]
```diff
@@ -1,4 +1,5 @@
 window.onload = ->
+  # this is a major security hole I think if it's accessible on the page
```
Comment (bbondy) [inline atom/renderer/lib/inspector.coffee]: Is this something we need to worry about?

**Comment by bbondy** [comment]
Great work.

I admittedly don't understand it all, so someone on the electron team would be a better reviewer when you upstream.  You might get some pushback on adding tests for this if it's not already covered by existing tests they have. 

I'm good for you to merge here for now though and just modify the commit(s) or do a new comment with changes they suggest upstream.

File: atom/renderer/lib/inspector.coffee (hunk 5 above)
Comment (bridiver) [inline atom/renderer/lib/inspector.coffee]: I need to test to make sure the inspector is only available in the dev tools and not the page. If the page can access it then it's definitely something we need to worry about

**Pull request merged**